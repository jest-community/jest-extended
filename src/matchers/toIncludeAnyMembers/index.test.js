import matcher from './';

expect.extend(matcher);

describe('.toIncludeAnyMembers', () => {
  const shallow = { hello: 'world' };
  const deep = { message: shallow };

  it('passes when given array is a subset', () => {
    expect([2, 3]).toIncludeAnyMembers([1, 2, 3]);
  });

  it('passes when given array contains some of the members', () => {
    expect([1, 2, 3]).toIncludeAnyMembers([2, 3, 4]);
  });

  it('passes when given array contains object value', () => {
    expect([shallow]).toIncludeAnyMembers([shallow, deep]);
  });

  it('fails when given array does not contain any members ', () => {
    expect(() => expect([4, 5, 6]).toIncludeAnyMembers([1, 2, 3])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given object is not an array', () => {
    expect(() => expect(7).toIncludeAnyMembers([1, 2, 3])).toThrowErrorMatchingSnapshot();
  });

  it('fails when expected object does not contain array value', () => {
    expect(() => expect([1, 2, 3]).toIncludeAnyMembers('world')).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toIncludeAnyMembers', () => {
  const shallow = { hello: 'world' };

  it('passes when given array does not contain primitive value', () => {
    expect(['hola', 'mundo']).not.toIncludeAnyMembers(['hello', 'world']);
  });

  it('passes when given array does not contain object value', () => {
    expect([{ hello: 'world' }]).not.toIncludeAnyMembers([{ world: 'hello' }]);
  });

  it('passes when given object is not array', () => {
    expect(7).not.toIncludeAnyMembers([7]);
  });

  it('fails when given object contains primitive value', () => {
    expect(() => expect([7]).not.toIncludeAnyMembers([7, 8])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given array contains object value', () => {
    expect(() => expect([{ foo: 'bar' }]).not.toIncludeAnyMembers([{ foo: 'bar' }])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given array contains array value', () => {
    expect(() => expect([[shallow]]).not.toIncludeAnyMembers([[shallow], 7])).toThrowErrorMatchingSnapshot();
  });
});
