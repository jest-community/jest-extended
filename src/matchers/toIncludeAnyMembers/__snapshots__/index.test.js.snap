// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toIncludeAnyMembers fails when given array contains array value 1`] = `
"<dim>expect(</><red>received</><dim>).not.toIncludeAnyMembers(</><green>expected</><dim>)</>

Expected list to not include any of the following members:
  <green>[[{\\"hello\\": \\"world\\"}], 7]</>
Received:
  <red>[[{\\"hello\\": \\"world\\"}]]</>"
`;

exports[`.not.toIncludeAnyMembers fails when given array contains object value 1`] = `
"<dim>expect(</><red>received</><dim>).not.toIncludeAnyMembers(</><green>expected</><dim>)</>

Expected list to not include any of the following members:
  <green>[{\\"foo\\": \\"bar\\"}]</>
Received:
  <red>[{\\"foo\\": \\"bar\\"}]</>"
`;

exports[`.not.toIncludeAnyMembers fails when given object contains primitive value 1`] = `
"<dim>expect(</><red>received</><dim>).not.toIncludeAnyMembers(</><green>expected</><dim>)</>

Expected list to not include any of the following members:
  <green>[7, 8]</>
Received:
  <red>[7]</>"
`;

exports[`.toIncludeAnyMembers fails when expected object does not contain array value 1`] = `
"<dim>expect(</><red>received</><dim>).toIncludeAnyMembers(</><green>expected</><dim>)</>

Expected list to include any of the following members:
  <green>\\"world\\"</>
Received:
  <red>[1, 2, 3]</>"
`;

exports[`.toIncludeAnyMembers fails when given array does not contain any members  1`] = `
"<dim>expect(</><red>received</><dim>).toIncludeAnyMembers(</><green>expected</><dim>)</>

Expected list to include any of the following members:
  <green>[1, 2, 3]</>
Received:
  <red>[4, 5, 6]</>"
`;

exports[`.toIncludeAnyMembers fails when given object is not an array 1`] = `
"<dim>expect(</><red>received</><dim>).toIncludeAnyMembers(</><green>expected</><dim>)</>

Expected list to include any of the following members:
  <green>[1, 2, 3]</>
Received:
  <red>7</>"
`;
